### PR TITLE
feat(form-components): label slot to prop update & new tooltip slot

### PR DESCRIPTION
**Pull request by jaideep08**
## Summary

Made the following changes:
- Checkbox Group
  1. Moved `label` to _prop_ from _slot_.
  2. Added `tooltip` _slot_.
- Dropdown
  1. Moved `label` to _prop_ from _slot_.
  2. Added `tooltip` _slot_.
- Toggle Button
  1. Added `tooltip` _slot_.
  2. Added `label` _prop_.

## ADO Story or GitHub Issue Link

[TASK 1931306](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1931306)
[TASK 1931309](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1931309)
[TASK 1931315](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1931315)

Bug: [BUG 1968192](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1968192)

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [x] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

## Screenshots
Screenshot 2024-10-17 at 1 05 40 PM
Screenshot 2024-10-17 at 1 06 38 PM
Screenshot 2024-10-17 at 1 06 44 PM
Screenshot 2024-10-17 at 1 06 54 PM
Screenshot 2024-10-17 at 1 07 12 PM
Screenshot 2024-10-17 at 1 07 36 PM

Screenshot 2024-10-18 at 4 52 56 PM
Screenshot 2024-10-18 at 4 53 07 PM
image

### CODE REVIEW

File: src/components/reusable/checkbox/checkboxGroup.ts [hunk 1]
```diff
@@ -149,7 +154,8 @@ export class CheckboxGroup extends FormMixin(LitElement) {
                   </abbr>
                 `
               : null}
-            <slot name="label"></slot>
+            <span>${ifDefined(this.label)}</span>
```
Comment (brian-patrick-3) [inline src/components/reusable/checkbox/checkboxGroup.ts:157]: You don't need to use `ifDefined` here, label is required.

File: src/components/reusable/toggleButton/toggleButton.ts [hunk 2]
```diff
@@ -2,16 +2,21 @@ import { LitElement, html } from 'lit';
 import { customElement, property, query } from 'lit/decorators.js';
 import { FormMixin } from '../../../common/mixins/form-input';
 import ToggleButtonScss from './toggleButton.scss';
+import { ifDefined } from 'lit/directives/if-defined.js';
 
 /**
  * Toggle Button.
  * @fires on-change - Captures the change event and emits the selected value and original event details.
- * @slot unnamed - Slot for label text.
+ * @slot unnamed - Slot for tooltip.
```
Comment (brian-patrick-3) [inline src/components/reusable/toggleButton/toggleButton.ts:10]: Name the slot "tooltip" to be consistent with all the other form components.

File: src/stories/migration.mdx [hunk 3]
```diff
@@ -14,6 +14,21 @@ Breaking Changes
...
+    - `label` slot moved to `label` prop
+  - Dropdown
+    - `label` slot moved to `label` prop
+- Added component slots
+  - Checkbox Group
+    - `tooltip` slot is added
+  - Dropdown
+    - `tooltip` slot is added
+  - Toggle Button
+    - `unnamed` slot is for tooltip
```
Comment (brian-patrick-3) [inline src/stories/migration.mdx:31]: Pull in the latest from main containing Vijeta's changes, and add your notes under the section she created.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
I don't see any error icon changes.

**Comment by jaideep08** [comment]
The error icon was already present for checkbox group and dropdown.

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by srpriyankashetty** [comment]
Single searchable is not showing the error message when made mandatory

**Comment by jaideep08** [comment]
> Single searchable is not showing the error message when made mandatory

Fixed it for both single searchable and multi-select searchable

**Review by kumarivijeta — APPROVED** [review]

**Review by kumarivijeta — COMMENTED** [review]
The info icon looks larger compared to the label text. Please set the info icon size to 16 to get proper alignment.

**Review by srpriyankashetty — COMMENTED** [review]
Verified the Checkbox, Dropdown and Toggle button - info icon and the tried changing the label name - looks fine.

**Pull request merged**